codes/c/chapter_heap/my_heap.c

```diff
@@ -0,0 +1,179 @@
+/**
+ * File: my_heap.c
+ * Created Time: 2023-01-15
+ * Author: Reanon (793584285@qq.com)
+ */
+
+#include "../include/include.h"
+
+#define MAX_SIZE 5000
+
+// 大顶堆
+typedef struct maxHeap {
+    // size 代表的是实际元素的个数
+    int size;
+    // 使用预先分配内存的数组，避免扩容
+    int data[MAX_SIZE];
+} maxHeap;
+
+void siftDown(maxHeap *h, int i);
```

Comment:
Do we need it?

Reply:
Yes.A function can't be called before defining it in C. If not  declared here, we should move "newMaxHeap" below "siftDown".